.github/workflows/deploy.yml

```diff
@@ -0,0 +1,94 @@
+name: CI/CD Pipeline
+
+on:
+  pull_request:
```

Comment:
please carefully check on part of action will trigger in main branch when merged

Reply:
for this one if it merge to main it will not run workflow.If we need to auto run this workflow after it merge we need to add

```
  push:
    branches:
      - main
```

but it will run this workflow after we push something to main branch. I think we can set cannot push code to main branch if we do like this

summary
we have 2 option
1. run workflow by ourself after merge.
2. set it run workflow auto but we need to set main branch to cannot push code.

Reply:
I think add push action for main and also add rule to block pushing for main branch is good pratice na.